docs/canary_releases.md

```diff
@@ -0,0 +1,70 @@
+# Canary releases
+
+There is a need to use canary releases in order to make sure newer versions
+of the gathering conditions don't break the Insights Operator gatherer.
+
+This is how a canary release would look like:
+
+```mermaid
+sequenceDiagram
+    actor RT as Rules Team
+    actor PT as Processing Team
+    participant GC as Gathering Conditions
+    participant AI as App Interface
+    participant GCS as Gathering Conditions Service
+    participant Unleash
+    participant Storage
+    actor C as Cluster
+    actor CC as Canary Cluster
+
+    Note left of GCS: Latest version is 1.0.1
```

Comment:
Let's express this consistently in terms of the two variables to avoid ambiguity. I suppose this means that the deployment uses `STABLE_VERSION = 1.0.1` and `CANARY_VERSION` is not set or is empty. This would answer @ikerreyes's [question](https://github.com/RedHatInsights/insights-operator-gathering-conditions-service/pull/234/files#r1675730125).

Reply:
We might also state the prerequisite here that the canary population in Unleash is supposed to be at 0% at this point.

Reply:
Updated